v2/internal/message.go

```diff
@@ -41,6 +43,19 @@ type Message struct {
 
 	// Other is the content of the message for the CLDR plural form "other".
 	Other string
+
+	// justCreated is used internally, to know whether the message ID
+	// has to be set (creation) or prepended with prefix (recursive call)
+	justCreated bool
```

Comment:
Is there any way to solve this problem without needing to add an unexported field to this struct?

Reply:
Did it! I agree, this was not very smart… I managed to use the `isMessage()` function instead for the same purpose.

This one is costly, so I anticipated the call (and pass the result as a parameter to `recGetMessages`) to avoid computing it twice for each submap.

While I was at it, I also factorized some duplicate code, this will make these functions easier to read and maintain, and at the same time improving coverage (reducing the content of the untested `switch` branches to trivial code).